src/coprocessor/statistics/analyze.rs

```diff
@@ -73,27 +77,48 @@ impl<S: Snapshot> AnalyzeContext<S> {
         Ok(res_data)
     }
 
-    // handle_index is used to handle `AnalyzeIndexReq`,
+    // batch_handle_index is used to handle `AnalyzeIndexReq`,
     // it would build a histogram and count-min sketch of index values.
-    fn handle_index(
+    fn batch_handle_index(
         req: AnalyzeIndexReq,
-        scanner: &mut IndexScanExecutor<TiKVStorage<SnapshotStore<S>>>,
+        scanner: &mut BatchIndexScanExecutor<TiKVStorage<SnapshotStore<S>>>,
     ) -> Result<Vec<u8>> {
         let mut hist = Histogram::new(req.get_bucket_size() as usize);
         let mut cms = CmSketch::new(
             req.get_cmsketch_depth() as usize,
             req.get_cmsketch_width() as usize,
         );
-        while let Some(row) = scanner.next()? {
-            let row = row.take_origin()?;
-            let (bytes, end_offsets) = row.data.get_column_values_and_end_offsets();
-            hist.append(bytes);
-            if let Some(c) = cms.as_mut() {
-                for end_offset in end_offsets {
-                    c.insert(&bytes[..end_offset])
+
+        let mut is_drained = false;
+        let mut batch_size = BATCH_INITIAL_SIZE;
+        while !is_drained {
```

Comment:
I guess there is no need to follow a grow strategy for analyze

Reply:
> I guess there is no need to follow a grow strategy for analyze

So is there a suitable constant for `batch_size`?